### DISCUSSION TITLE
WinForms TextBox Caret does not blink

### DISCUSSION THREAD

**Discussion by dakusan** [dotnet/winforms#6185] [Q&A] [answered]
I'm using .NET Core 3.1, but another user on StackOverflow reported the same problem in .NET 5. And one user reported “Mine blinks 5-6 times then stops blinking.”.

I have a System.Windows.Forms.TextBox that I added to the form through the designer in Visual Studio 2019 C# to a “Windows Forms App (.NET Core)” project. The caret just stays solid and does not blink. The Win32 APIs don't seem to help and I can't find anything on the System.Windows.Forms.TextBox or its parent classes that may help.

To reproduce doesn't require anything more than adding a TextBox onto a Form.

### Further technical details
dotnet --info
.NET Core SDK (reflecting any global.json):
 Version:   3.1.401
 Commit:    5b6f5e5005

Runtime Environment:
 OS Name:     Windows
 OS Version:  6.1.7601
 OS Platform: Windows
 RID:         win7-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.401\

Host (useful for support):
  Version: 3.1.7
  Commit:  fcfdef8d6b

.NET Core SDKs installed:
  2.1.701 [C:\Program Files\dotnet\sdk]
  3.1.401 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.21 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.7 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by RussKie** [comment, marked as answer]
This is a standard Windows behaviour. It is reproducible in .NET Framework as well:

https://user-images.githubusercontent.com/4403806/141871888-afb5f6fb-0bc2-46b5-8d00-474670d79b6e.mp4
